OfflineUX.md

```diff
@@ -0,0 +1,129 @@
+If you’re making a new feature, think about whether any data would need to be retrieved or stored from anywhere other than the local device. If data needs to be stored or retrieved, then the answer is YES.
+
+2. What type of request is being made?
+
+This question can be answered by thinking about what’s happening in the server. If there’s new data being saved on the server, you’re making a WRITE request. If you’re retrieving existing data from the server, you’re making a READ request. 
+
+3. Do we need to show up-to-date data?
+
+You should answer yes to this question if it’s important that the user sees current data/content. An example is viewing a chat. Even though the user won’t be able to see new chat messages, it’s still important for us to show the user the old ones.
+
+4. Is the UI a form?
+
+Any easy way to tell if something is a form is to try and find a submit button. If a submit button is present or if the user is filling out fields that we want to save locally, answer YES to this question.
+
+5. Can the server response be anticipated?
+
+Assuming that the request succeeds when it’s sent to the server, do you know what would happen next? For example, we would answer NO if there is some new data coming back from the server that we have no way of guessing. Answer YES if it’s easy to tell what the response from the server would be. 
+
+6. Is there validation done on the server?
+
+If there is some validation happening on the server that needs to happen before the feature can work, then we answer YES to this question. Remember, this is referring to validation that cannot happen on the front end (e.g. min/max characters in a password). For example, if we want to set up a bank account then our answer to the previous question (5) is YES (we could imagine that the server returns a 200 and we would know what the next page would look like). However, our answer to this question is NO (because we can’t suggest to the user that their request succeeded when really it hasn’t been sent yet–their card wouldn’t work!)
+
+This question can be tricky, so if you’re unsure, please ask a question in #expensify-open-source and tag the contributor management engineer team.
+
+7. Does the user need to know if the action was successful?
+
+Think back to the pinning example from above. In this case, the user doesn’t need to know that their pinned reports NVP has been updated. To them the impact of clicking the pin button is that their chat is at the top of the LHN. It makes no difference to them if the server has been updated or not, so the answer would be NO. Now let’s consider the case of sending a comment. In this example, the user needs to know if their comment was actually sent, because they may need to know if the person they sent it to can read it. In this case our answer is YES.
+
+### Descriptions of the UX Patterns
```

Comment:
We'll want each of these sections to link to the refactored commands we're writing from the Optimized API doc.

Reply:
Are we holding on this or going back later to update?

Reply:
I think we should come back later to update this. Once we refactor each example command, we can write the `How to implement` section with more ease (since we'll be able to reference code directly).